File: HW1/HW1CodesSolution/HW1ForChecking.py
import HW1functions as HW1f
import numpy as np

# input
L =  250 # total length of the bar
Nelem = 4 # number of elements
F =  1000# applied force on the final nodes
E =  70e3 # elastic modulus 
BC = np.array([[1],[0]]) #form: [node numbers], [applied displacements]

# functions here
u, strain, stress = HW1f.calcDisp(L, Nelem, F, E, BC)

# output
for el in range(Nelem):
	print ("the stress in element ", (el+1) , " equals ", "%.2f" % stress[el], "MPa")
	
print("The displacement of the final node is ", "%.4f" % u[-1], "mm")

File: HW1/HW1CodesSolution/HW1ForChecking.m
clc;
clear all;
close all;

% input
L =  250; % total length of the bar
Nelem = 4; % number of elements
F =  1000; % applied force on the final nodes
E =  70e3; % elastic modulus 
BC = [1; 0]; %form: [node numbers; applied displacements]

% functions here
[u, strain, stress] = HW1functions(L, Nelem, F, E, BC);

% output
for el=1:Nelem
	fprintf(['the stress in element ', num2str(el) , ' equals ', num2str(stress(el)), ' MPa \n'])
end
	
fprintf(['The displacement of the final node is ', num2str(u(end)),' mm \n'])